### DISCUSSION TITLE
Custom CA bundle for Python Algorithms?

### DISCUSSION THREAD

**Discussion by agilmore2** [opendcs/opendcs#436] [Q&A]
We have Python computations that make Web API calls. Recent SSL certs are no longer trusted. How do we tell OpenDCS python requests library to use an alternate CA bundle? Other uses of python requests succeed at this using the environment variable REQUESTS_CA_BUNDLE. We'll test this, but in case, what would be another alternative? Updating the python requests library certs.py script inside OpenDCS seems clunky.

**Comment by MikeNeilson** [comment]
Is it actually using the bundled certs or the JVM certs?

**Comment by agilmore2** [comment]
The Python environment includes requests and "certifi" which is a global CA trust bundle. I'd expect the requests calls use the bundled CA info. We tried setting the requests ca verify parameter to the path of the system bundle, but that still failed. Setting verify=False worked, but that's not the greatest.
